### ISSUE TITLE
Some Suggestions

### ISSUE THREAD

**Issue by LePanda026** [FrancescoSaverioZuppichini/ViT#5]
Hello, I am a beginner, thanks for youer torch-version vit.
After reading your code, I noticed that there seems to be an issue with this line of code：
**att = F.softmax(energy, dim=-1) / scaling**
It seems that the division by scaling should be done before the softmax.
Additionally, I believe that having the code alone is not sufficient; could you please realise the the complete pipeline?

**Comment by FrancescoSaverioZuppichini** [comment]
Hi @LePanda026 you are correct, the code alone is just the model itself I don't have a pipeline
